text/0003-monorepo.md

```diff
@@ -0,0 +1,76 @@
+- Start Date: 2020-07-03
+- Implementation PR: (leave this empty)
+- Store Discussion Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes a solution to create a single repository to all apps of the Store Framework. 
+
+# Basic example
+
+We want to do something similar that the following frameworks do:
+
+* [Angular](https://github.com/angular/angular)
+* [React](https://github.com/facebook/react)
+* [Material UI](https://github.com/mui-org/material-ui)
+
+They keep all packages in the same repository
+
+# Motivation
+
+When the Store Framework started it had only one team united team with all the people in the same place and it had not so many apps (around 12 apps) to this team to support, so it was easy to control who was doing what and if it was being done respecting the standards of the framework, but today we have around 73 repositories that 6-7 different teams of VTEX support and on top of it now we have an open-source community that uses those apps and even contribute to it. With all this code base it became complicated to update tooling, fix technical depts, and even update these apps to respect best practices of code or API. With a monorepo we can get closer to the idea of creating a product that has the same ideas across all blocks.
+
+Pain points of keeping 73 repos up to date:
+
+* Lint rules always get out of date in some forgotten apps
+* Github templates have the same problem of lint rules
+* API patterns and best practices to develop a new block
+* Avoid duplication of logic
+* Test setup
+* Misleading open-source flow to create issues
+
+
+# Detailed design
+
+Since this is not a feature but an organizational change that might help everyone that works in Store Framework and the open-source community this we don't have much to add to this section, but we expect to fix all the problems listed above and that the monorepo help us to make it easier the following topics:
+
+* Share code to make testing easier
+* Create a calendar of support for majors
+* Plan new majors for the whole framework
+* Create migration guides to each change of these majors
+* Predictable manual updates to our clients to get a better store
+
+All these points about majors should have an RFC to explain better how we should do it.
+
+# Drawbacks
+
+Why should we *not* do this? Please consider:
+
+- To launch a new major we would need to launch a major of all apps or to put the legacy major in a separate folder and to add the current in the place of the current major (similar with what the [search-result](https://github.com/vtex-apps/seach-result) does)
+- Older major versions that were being kept in different branches would have to make the same strategy to continue to exist
+- We would have to update the bot that launches a new version of the apps
+- The repository would be *incredibly* big
```

Comment:
Is not like that, but people should take an extra care with git now because a bad rebase might create a lot of problems. And the git commands might become a bit slower (no so much)